File: challenges/codeeval/easy/111/b1nary.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            string[] words = line.Split();
            string longest = words[0];
            foreach (string word in words)
                if (word.Length > longest.Length) {
                    longest= word;
                }
            Console.WriteLine(longest);
        }
    }
}

File: challenges/codeeval/easy/111/b1nary.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            String[] words = line.split(" ");
            String longest = words[0];
            for (int i=0; i<words.length; i++){
                if (words[i].length() > longest.length()){
                    longest = words[i];
                }
            }
            System.out.println(longest);
        }
    }
}
